ADR/0033-provisioning-test-resources.md

```diff
@@ -0,0 +1,277 @@
+# 33. Provisioning Clusters for Integration Tests
+
+Date: 2024-03-07
+
+## Status
+
+Accepted
+
+Supersedes:
+
+- [ADR 08. Environment Provisioning](0008-environment-provisioning.html)
+- Environment provisioning parts of [ADR 32. Decoupling Deployment](0032-decoupling-deployment.html)
+
+## Context
+
+This decision clarifies how integration test environments will be dynamically provisioned. In prior
+decisions it was believed that [Dynamic Resource Allocation] (DRA) would graduate out of OpenShift
+TechPreview on a timeline suitable for this project. This is no longer the case and, as such, a new
+approach to requesting compute resources for test pipelines is needed. DRA is still expected to
+become the kubernetes-native approach to managing dynamically provisioned resources across pods.
+Therefore, any interim solution should intentionally avoid introducing new barriers that might
+prevent the adoption of DRA some day.
+
+The problem of provisioning test resources/environments can be broken down into a few questions:
+
+1. How can resources (OpenShift clusters, to start) be provisioned, efficiently, without exposing
+   shared cloud/infra credentials to the end user?
+1. In case the user requires more customization than what's provided by shared configuration, how
+   can they provide their own (including infra credentials)?
+1. How does the end user request resources from an integration `Pipeline`?
+
+Provisioning of infrastructure can consume some significant compute resources itself. All possible
+solutions must also account for the challenge of scalability in production.
+
+### Cluster Provisioning
```

Comment:
Can you introduce this section briefly before going into the various alternatives just to help the reader as they progress through them? What are the criteria for inclusion in these provisioning technologies?

Reply:
I added a brief intro, PTAL.